packages/flutter/lib/src/widgets/restoration_properties.dart

```diff
@@ -185,6 +228,35 @@ class _RestorablePrimitiveValue<T extends Object> extends RestorableValue<T> {
   }
 }
 
+// _RestorablePrimitiveValueN and its subclasses allows null values.
+class _RestorablePrimitiveValueN<T extends Object> extends RestorableValueN<T> {
```

Comment:
I believe you can revert all changes you made to `widgets/restoration.dart` and above this line in this file by just changing this line to:

```suggestion
class _RestorablePrimitiveValueN<T extends Object?> extends RestorableValue<T> {
```

The `Object?` ensures that T can be null, if necessary.

After you do that, a lot of `?` below in this file can also be removed, I think.

Reply:
(With that in place, _RestorablePrimitiveValue can probably share some implementation with _RestorablePrimitiveValueN by just making _RestorablePrimitiveValue extend _RestorablePrimitiveValueN.)

Reply:
Done!